src/game/game_state.cpp

```diff
@@ -50,7 +60,13 @@ void GameState::update() {
     scene_manager.get_current_scene()->update();
 
     if (curr_phase == &dungeon_phase) {
```

Comment:
Yes, I agree - though I didn't modify it further since this is dependent on upcoming `GameState` and `Phase` refactoring.  Right now phases' update function don't take in any game state/context; this is an example of something that would go there.